### PR TITLE
Delete CLSID_ComCallUnmarshal and related code

### PR DESCRIPTION
**Pull request by jkotas**
Not used in CoreCLR

### CODE REVIEW

File: src/coreclr/src/vm/reflectioninvocation.cpp [hunk 1]
```diff
@@ -994,9 +994,6 @@ FCIMPL5(Object*, RuntimeMethodHandle::InvokeMethod,
     }
 #endif
 
-    // Skip the activation optimization for remoting because of remoting proxy is not always activated.
```
Comment (jkotas) [inline src/coreclr/src/vm/reflectioninvocation.cpp]: Small unrelated cleanup

File: src/coreclr/src/vm/reflectioninvocation.cpp [hunk 2]
```diff
@@ -1322,11 +1316,7 @@ FCIMPL5(Object*, RuntimeMethodHandle::InvokeMethod,
     if (fConstructor)
     {
         // We have a special case for Strings...The object is returned...
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/src/vm/reflectioninvocation.cpp]: Is this comment still valid? Seems a but disconnected with the updated code.

**Review by AaronRobinsonMSFT — APPROVED** [review]

File: src/coreclr/src/vm/reflectioninvocation.cpp (hunk 2 above)
Comment (jkotas) [reply src/coreclr/src/vm/reflectioninvocation.cpp]: Yes, this comment is still valid. We will only get here for strings.
